Lib/test/test_ast.py

```diff
@@ -1334,6 +1334,51 @@ def test_dump_incomplete(self):
             "Raise(cause=Name('e', Load()))"
         )
 
+    def test_dump_show_empty_false(self):
+        self.assertEqual(
+            ast.dump(ast.arguments(), show_empty=False),
+            "arguments()",
+        )
+
+        self.assertEqual(
+            # Corner case: there are no real `Name` instances with `id=''`:
+            ast.dump(ast.Name(id='', ctx=ast.Load()), show_empty=False),
+            "Name(ctx=Load())",
+        )
+
+        def check_text(code, expected, **kwargs):
+            nodes = ast.parse(code)
+            self.assertEqual(
+                ast.dump(nodes, show_empty=False, **kwargs),
+                expected,
+            )
+
+        check_text(
```

Comment:
Can you add a test involving None? I think we may need to add a special case so it shows `ast.Constant(value=None)` instead of just `ast.Constant()`:

```
>>> ast.Constant()
<stdin>-3:1: DeprecationWarning: Constant.__init__ missing 1 required positional argument: 'value'. This will become an error in Python 3.15.
  ast.Constant()
<ast.Constant object at 0x103880c90>
```